[X86-64] Add support for CALL64m instruction

martin-fink opened the pull request:
This adds support for indirect calls from memory locations
Example:

```s
call [rsp]
```

I have moved some functionality from `raiseMemRefMachineInstr` into it's own function `X86MachineInstructionRaiser::getMemoryRefValue`, since that functionality is needed to load the memory value stored in `CALL64m`'s operand.

bharadwajy commented:
Thanks! 

Unfortunately, the added test fails for me on my Ubuntu 20.04. See two attached files for additional info.

1. [raise-call64m-dis.txt](https://github.com/microsoft/llvm-mctoll/files/6475624/raise-call64m-dis.txt) - Raised IR
2. [PR-90.txt](https://github.com/microsoft/llvm-mctoll/files/6475625/PR-90.txt) - verbose compilation output of raised IR

martin-fink commented:
I found the issue, I've pushed the change with the fix.

I removed the push instruction in the test:

```s
mov rax, OFFSET func1
push rax
call [rsp]
```

And replaced it with an explicit store to the stack:

```s
mov rax, OFFSET func1
sub rsp, 8
mov [rsp], rax
call [rsp]
```

The explicit move now stores the correct value and `call [rsp]` is able to do it's thing.

bharadwajy commented:
Thanks for narrowing the issue. That points to a bug in raising the push instruction. I fixed that bug.

I pushed your original changes along with my bug fix.
